mcstatus/motd.py

```diff
@@ -0,0 +1,545 @@
+    bedrock: bool = False
+
+    @classmethod
+    def parse(cls, raw: RawResponseDescription, *, bedrock: bool = False) -> Self:
+        """Parse a raw MOTD to less raw MOTD (``Motd.parsed`` field).
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition? Nothing changes here, just sets attribute.
+        :returns: New ``MOTD`` instance.
+        """
+        if isinstance(raw, list):
+            raw = RawResponseDescriptionWhenDict(**{"extra": raw})
+
+        if isinstance(raw, str):
+            parsed = cls._parse_as_str(raw, bedrock=bedrock)
+        else:
+            parsed = cls._parse_as_dict(raw, bedrock=bedrock)
+
+        return cls(parsed, bedrock)
+
+    @classmethod
+    def _parse_as_dict(
+        cls, item: RawResponseDescriptionWhenDict, *, bedrock: bool = False, auto_add: list[ParsedMotdItem] | None = None
+    ) -> list[ParsedMotdItem]:
+        """Parse a MOTD when it's dict.
+
+        :param item: ``dict`` directly from the server.
+        :param bedrock: Is the server Bedrock Edition?
+            Nothing does here, just going to ``_parse_as_str`` while parsing ``text`` field.
+        :param auto_add: Values to add on this item.
+            Most time, this is ``Formatting`` from top level.
+        :returns: ``ParsedMotdItem`` list, which need to be passed to ``__init__``.
+        """
+        parsed_motd: list[ParsedMotdItem] = auto_add if auto_add is not None else []
+
+        if "color" in item:
+            try:
+                parsed_motd.append(MinecraftColor[item["color"].upper()])
+            except KeyError:
+                if item["color"] == "reset":
+                    # minecraft server actually can't return {"reset": True}, instead
+                    # it will return {"color": "reset"}. but for follow naming, we
+                    # will not move `Formatting.RESET` to `MinecraftColor`.
+                    #
+                    # see https://wiki.vg/Chat#Shared_between_all_components, `color` field
+                    parsed_motd.append(Formatting.RESET)
+                else:
+                    parsed_motd.append(WebColor.from_hex(item["color"]))
+
+        for style_key, style_val in Formatting.__members__.items():
+            lowered_style_key = style_key.lower()
+            if item.get(lowered_style_key) is False or item.get(lowered_style_key) == "false":
+                try:
+                    parsed_motd.remove(style_val)
+                except ValueError:
+                    # some servers set the formatting keys to false here, even without it ever being set to true before
+                    continue
+            elif item.get(lowered_style_key) is not None:
+                parsed_motd.append(style_val)
+
+        if (text := item.get("text")) is not None:
+            parsed_motd.extend(cls._parse_as_str(text, bedrock=bedrock))
+        if text is None and (translate := item.get("translate")) is not None:
```

Comment:
If I correctly understand docs, it can't contain `text` and `translate` items in one time.

![Screenshot_2022-12-13-09-50-42-624-edit_com opera gx](https://user-images.githubusercontent.com/68118654/207270366-54860991-91b1-4c58-a9f2-3c479b1e3a49.jpg)

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/ab68edac24159f7185f462310971171426167a68..676680ad8a2d0f97a9421da968add8149544a19c).